### ISSUE TITLE
Witness discount

### ISSUE THREAD

**Issue by shesek** [tothemoon-org/extension-blocks#8]
Can you please describe the motivation behind removing the witness discount that helps better align the incentives for enlarging the UTXO set?

**Comment by chjj** [comment]
The witness scale factor has been removed on the consensus layer, since it's not necessary for calculating DoS limits anymore. 

That being said, policy is still up in the air. There will be a second document for "implementation commentary", discussing guidelines for policy.

**Comment by afk11** [comment]
The witness scale factor _is_ policy; there's no way to detect if a miner hasn't offered a discount on witness data unless you look at the fee rate of the transactions being mined. 

Any miner can say "I would like to be more competitive" and offer some discount on _some_ data (as opposed to lowering their absolute fee rate). 

I'd be very curious what you have to say about preventing this.

**Comment by chjj** [comment]
@afk11, no. Segwit's witness discount policy is a _specific_ side effect of segwit's consensus rules. It's not arbitrary. Fee policy is greatly affected by the economic incentives that consensus rules create: if consensus DoS limits treat witness data as adding less weight, that will likely be taken into account on the policy layer.

Since extblk consensus rules regarding DoS limits weren't really solidified before, there was no way to tell whether a witness discount would be available or even make sense on the policy layer. Going by segwit-style weight makes little sense in terms of consensus rules for extblk, since the weight metric segwit uses is partly a hack to add more capacity to the main block. That metric isn't necessary for the extension block. All that said, extblk's inputs cost could be weighed against a lesser factor than outputs cost.

> Any miner can say "I would like to be more competitive" and offer some discount on some data (as opposed to lowering their absolute fee rate).

Sure, a miner could do anything they wanted and mine 0 fee transactions right now if they want. It's a moot point. If that's the conclusion, then there's no sense in concerning ourselves with any of this. Luckily, that's not usually what happens. Transaction selection is only ever reasonably determined by fee rate weighed against DoS limits.

**Comment by afk11** [comment]
Witness space is counted as 1/4 of the main space. Therefore, if you remove the discount (= charge more), then your available block space gets smaller. Removing the discount is safe, but it reduces the miners capacity.

**Comment by afk11** [comment]
Like if you double counted sigOps while producing blocks (not validation), you'd lower the number of transactions you could mine in a block, but still be compatible with the network.

If you didn't count sigops, you'd make yourself incompatible.

Removing the discount is similar to the first, it's basically lowering your weight back to 1 million.
